### ISSUE TITLE
Deployment instructions for Github pages no longer works. 

### ISSUE THREAD

**Issue by karmenlu** [sphinx-doc/sphinx#11413]
🤖 Rendered output: https://sphinx--9852.org.readthedocs.build/en/9852/tutorial/deploying.html

_Originally posted by @astrojuanlu in https://github.com/sphinx-doc/sphinx/issues/9852#issuecomment-973157631_

I saw from this closed MR that the Sphinx tutorial was updated with a working Github workflow yaml file. This Github yaml file is now failing for me. This rendered output link provided by @astrojuanlu opens a 404 page.

I'm new to Sphinx. My goal is to deploy documentation with a Github workflow but I'm unsure how to edit the `sphinx/yml` file included in the tutorial.

**Comment by jsdodge** [comment]
The `sphinx.yml` file included in the [documentation](https://www.sphinx-doc.org/en/master/tutorial/deploying.html#id5) did not work for me, either, although it looks like my reasons are a little different from the OP. The build failed, returning the following error:

```
/usr/bin/git push origin gh-pages
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/username/repository.git/': The requested URL returned error: 403
Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"
``` 


The [documentation](https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-first-deployment-with-github_token) for the GitHub Pages action suggested enabling write permissions as shown below, which worked. I recommend that the Sphinx documentation mention this, too.

```YAML
name: Sphinx build

on: push

jobs:
  build:
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
    - uses: actions/checkout@v3
    - name: Build HTML
      uses: ammaraskar/sphinx-action@master
    - name: Upload artifacts
      uses: actions/upload-artifact@v3
      with:
        name: html-docs
        path: docs/build/html/
    - name: Deploy
      uses: peaceiris/actions-gh-pages@v3
      if: github.ref == 'refs/heads/main'
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        publish_dir: docs/build/html
```

**Comment by AA-Turner** [comment]
Fixed by db739af
